enhancements/rhcos/split-rhcos-into-layers.md

```diff
@@ -0,0 +1,458 @@
+into a statically-typed language that can literally live in openshift/os and get compiled and
+added as part of the derived (now multi-staged) layered build.
+
+At a more practical level, this helps reduce git churn for both the RHCOS pipeline and the
+OCP layered build.
+
+#### Add more comprehensive CI
+
+Another important follow-up would to be to hook up better CI testing to openshift/os. There is
+currently no CI test on that repo which actually launches a cluster. The reason is that RHCOS
+is just built differently. But now that the node image is simply yet another layered image build,
+it fits perfectly in Prow's opinionated model of building the image and shoving it in the test
+release payload with which to launch test cluster.
+
+#### Re-organize MCO templates
+
+Currently, there's a pile of templated files that live in the MCO. These
+files configure systemd services and run extra logic as required by OCP and
+conceptually have nothing to do with the MCO itself. It would then make sense to
+move them to the openshift/os repo alongside the rest of the definition for the
+node image.
+
+In a first step, we would leave them in the MCO but work on "de-templatizing"
+them by leveraging existing systemd directives as much as possible (e.g.
+conditionalizing on platform can be done with `ignition.platform.id=azure`,
+conditionalizing on OVN could be done with a generator or by having it pulled in
+by another systemd unit, etc...).
+
+Once they're fully de-templatized, we can move them to openshift/os so that they
+are part of the node image proper.
+
+This would considerable shrink the amount of templates that the MCO would
+need to process, and as a result serve via the MCS and/or layer via on-cluster
+layering.
+
+### Risks and Mitigations
+
+#### Versioning scheme
+
+The versioning scheme for the *node image* will remain unchanged (e.g.
+418.94.202409090849-0). But the versioning scheme for the *bootimages* will
+change from e.g. 418.94.202409090849-0 to e.g. 9.4.202409090849-0 since they'll
+no longer be versioned with OpenShift. (But note that the format itself remains
+the same, and is still semver-compatible.)
+
+This change is likely to throw off anything that currently parses the version
+string to for example derive the OCP and RHEL versions for which an RHCOS image
+was built. Those will need to be adapted if they deal with bootimages. The
+release browser as mentioned above is a likely candidate.
+
+Additionally, RHCOS artifacts on the official mirrors are currently versioned
+by the OCP release (e.g. `rhcos-4.16.3-x86_64-live.x86_64.iso`) and kept under
+OCP-versioned directories (e.g. `4.16.3/`). We don't want customers to have to
+know which version of RHCOS bootimages to download for their target OCP version.
+So to avoid confusing customers, we should still have OCP-versioned directories
+but it would be wasteful to not symlink the bootimages on the server side to
```

Comment:
Unfortunately, mirror.openshift.com doesn't support symlinks.

Reply:
Ouch, that's unfortunate. Or ideally just any kind of content-addressed storage. But anyway, softened the wording there.